deepchecks/nlp/checks/data_integrity/under_annotated_segments.py

```diff
@@ -0,0 +1,290 @@
+        display_msg = f'Showcasing intersections of {self.segment_by} with most under annotated segments.<br> The ' \
+                      'full list of under annotated segments can be observed in the check result value. '
+        display_fig = self._generate_scatter_plot_display(encoded_dataset.features_columns, score_per_sample,
+                                                          text_data.text, weak_segments, cat_features, avg_score)
+        return CheckResult(value=check_result_value, display=[display_msg, display_fig])
+
+    @staticmethod
+    def _get_box_boundaries(feature_data: pd.Series, segment: Tuple[float, float]) -> Tuple[float, float]:
+        lower = segment[0] if segment[0] != -np.inf else np.nanmin(feature_data)
+        upper = segment[1] if segment[1] != np.inf else np.nanmax(feature_data)
+        return lower, upper
+
+    def _generate_scatter_plot_display(self, encoded_data: pd.DataFrame, is_annotated: pd.Series,
+                                       text: np.ndarray, weak_segments: pd.DataFrame,
+                                       cat_features: List[str], avg_score: float) -> DisplayMap:
+        display_tabs = {}
+        if weak_segments.shape[0] > self.n_to_show:
+            weak_segments = weak_segments.iloc[:self.n_to_show, :]
+        encoded_data['text'] = text
+
+        # virtual col is used when we have only one feature controlling the segment
+        jitter = 0.25
+        encoded_data['virtual_col'] = np.random.uniform(-jitter, jitter, len(encoded_data))
+
+        sampled_data = encoded_data.sample(MAX_SAMPLES_IN_FIGURE, random_state=42)
+        annotated_data = sampled_data[is_annotated == 1]
+        not_annotated_data = sampled_data[is_annotated == 0]
+        for _, row in weak_segments.iterrows():
+            fig = go.Figure()
+            feature_1, feature_2 = row['Feature1'], row['Feature2']
+            feature_1_lower, feature_1_upper = self._get_box_boundaries(encoded_data[feature_1], row['Feature1 Range'])
+            if feature_2 != '':  # segment by two features
+                feature_2_lower, feature_2_upper = self._get_box_boundaries(encoded_data[feature_2],
+                                                                            row['Feature2 Range'])
+                hover_template = feature_1 + ': %{x}<br>' + feature_2 + ': %{y}<br>text: %{text}<br>Annotated: '
+                tab_title = f'{feature_1} vs {feature_2}'
+                msg = f'Under annotated segment contains samples with {feature_1} in range ' \
+                      f'{[format_number(feature_1_lower), format_number(feature_1_upper)]} and {feature_2} in range ' \
+                      f'{[format_number(feature_2_lower), format_number(feature_2_upper)]}.'
+            else:  # segment by one feature
+                feature_2 = 'virtual_col'
+                feature_2_lower = encoded_data['virtual_col'].min() * 1.3
```

Comment:
0.25 * 1.3? Not better to have the final result in `jitter`?

Reply:
no, it a display mechanism